utils/tfhe-versionable/src/lib.rs

```diff
@@ -356,23 +394,23 @@ where
     }
 }
 
-impl<T: NotVersioned + Clone + Serialize + DeserializeOwned> Versionize for AVec<T> {
-    type Versioned<'vers> = &'vers [T] where T: 'vers;
+impl<T: VersionizeVec + Clone> Versionize for AVec<T> {
+    type Versioned<'vers> = T::VersionedSlice<'vers> where T: 'vers;
 
     fn versionize(&self) -> Self::Versioned<'_> {
-        self.as_slice()
+        T::versionize_slice(self.as_slice())
     }
 
-    type VersionedOwned = AVec<T>;
+    type VersionedOwned = T::VersionedVec;
 
     fn versionize_owned(&self) -> Self::VersionedOwned {
-        self.clone()
+        T::versionize_vec(self)
     }
 }
 
-impl<T: NotVersioned + Clone + Serialize + DeserializeOwned> Unversionize for AVec<T> {
+impl<T: UnversionizeVec + Clone> Unversionize for AVec<T> {
     fn unversionize(versioned: Self::VersionedOwned) -> Result<Self, UnversionizeError> {
-        Ok(versioned)
+        T::unversionize_vec(versioned).map(|value| AVec::from_iter(0, value))
     }
 }
 
```

Comment:
This commit "feat(versionable): Add support for Vec of custom types" could be moved before commit "feat(versionable): Add versionize support for aligned-vec types" (mayb better to merge them then)

Reply:
good idea !

Reply:
done: I reordered it and now the AVec impl is similar to the Vec one. I didn't merge the commits, since the aligned-vec commit adds a new dependency to the crate, I figured it should stand on its own.